### ISSUE TITLE
DuckDuckGo blocked

### ISSUE THREAD

**Issue by Pistos** [searx/searx#3441]
**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**

Running off `master` at commit 117dbd462f39fb4c

**How did you install Searx?**

Installed and run manually from source (not docker)

**What happened?**

After starting searx, the first search says in the UI that duckduckgo was blocked. The `debug: True` logging shows:

```
DEBUG:urllib3.connectionpool:https://lite.duckduckgo.com:443 "POST /lite HTTP/1.1" 403 None
DEBUG:urllib3.connectionpool:https://www.bing.com:443 "GET /search?q=language%3AEN+hello+world&first=1 HTTP/1.1" 200 None       
DEBUG:urllib3.connectionpool:https://search.yahoo.com:443 "GET /search?p=hello+world&ei=UTF-8&fl=1&vl=lang_en&btf=&fr2=time&age=&b=1&xargs=0 HTTP/1.1" 200 None                                                                                                 
DEBUG:searx:duckduckgo: ErrorContext('/home/searx/searx/searx/search/processors/online.py', 96, "response = req(params['url'], **request_args)", 'searx.exceptions.SearxEngineAccessDeniedException', None, ('HTTP error 403',))
ERROR:searx.search.processor.online:engine duckduckgo : Searx is blocked                                                        
Traceback (most recent call last):                                                                                              
  File "/home/searx/searx/searx/search/processors/online.py", line 144, in search                                               
    search_results = self._search_basic(query, params)                                                                          
  File "/home/searx/searx/searx/search/processors/online.py", line 124, in _search_basic                                        
    response = self._send_http_request(params)                                                                                  
  File "/home/searx/searx/searx/search/processors/online.py", line 96, in _send_http_request
    response = req(params['url'], **request_args)                                                                               
  File "/home/searx/searx/searx/poolrequests.py", line 223, in post                                                             
    return request('post', url, data=data, **kwargs)                                                                            
  File "/home/searx/searx/searx/poolrequests.py", line 202, in request                                                          
    raise_for_httperror(response)                                                                                               
  File "/home/searx/searx/searx/raise_for_httperror.py", line 62, in raise_for_httperror
    raise SearxEngineAccessDeniedException(message='HTTP error ' + str(resp.status_code),
searx.exceptions.SearxEngineAccessDeniedException: HTTP error 403, suspended_time=86400
```

as per https://github.com/searx/searx/blob/117dbd462f39fb4cc2f12e9f7b9acf8d3fff419b/searx/search/processors/online.py#L195-L198

The block message no longer appears in the UI after the first search. I'm assuming searx recognizes the block and enacts a one-day (86400-second) cooldown.

**How To Reproduce**

Of course, this is specific to my instance, which I am not willing to share. Nevertheless, I hope to be able to resolve it with some dialogue.

**Expected behavior**

DDG is not blocked

**Screenshots & Logs**

See above.

**Additional context**

Is there anything I can do to lift the block? Should I disable the DDG engine for a few days?

**Comment by unixfox** [comment]
Related issue on searxng: https://github.com/searxng/searxng/issues/1854

**Comment by Pistos** [comment]
Thank you. Can confirm that adding a trailing slash to https://github.com/searx/searx/blob/117dbd462f39fb4cc2f12e9f7b9acf8d3fff419b/searx/engines/duckduckgo.py#L48 makes DDG work again. Thanks.

**Comment by cyclaero** [comment]
Just another problem emerged with duckduckgo.py, see searxng/searxng#2080

In addition to the trailing slashes, also the following needs to be added on line 76 of searx/searx/engines/duckduckgo.py
`params['headers']['Referer'] = 'https://lite.duckduckgo.com/'`

With that, duckduckgo is responding via searx again. I did the changes on my fork, however mine contains several other additions, and I am looking already for an hour or so, on how to send a pull request for the duckduckgo fixes without all my other changes as well. Can this be done? Then a pointer would be nice.

In case not, please may I ask the maintainer to do these minor code changes, which are already confirmed by searxng.

Best regards

Rolf

**Comment by cyclaero** [comment]
Never mind, I found a way to send the pull request, s. above.

**Comment by Pistos** [comment]
@cyclaero Thank you. That fixes it for me too.

**Comment by kvch** [comment]
Closing

**Comment by Pistos** [comment]
For what it's worth, DDG is blocked again in my instance. I guess they keep changing their rules.

**Comment by eggercomputerchaos** [comment]
same for me :-(

**Comment by eggercomputerchaos** [comment]
does anyone tried this: 
https://github.com/searxng/searxng/commit/9faf0dc2891e1509861e0be9ff9db67f833f8d82

**Comment by eggercomputerchaos** [comment]
for me 
searxng/searxng@9faf0dc
works 
- how long? don't know

**Comment by Pistos** [comment]
Blocked again. Do they not want to allow Searx to get DDG results?

**Comment by Pistos** [comment]
This is fixed for me in latest (75b859d2a897).
